starlette/staticfiles.py

```diff
@@ -85,6 +100,10 @@ def not_modified_response(self, stat_headers: dict) -> Response:
         if self.check_directory is not None:
             await self.check_directory_configured_correctly()
 
+        (head, tail) = os.path.split(self.path)
+        if self.serve_index_html and tail == ".":
```

Comment:
I'm confused by the `tail == "."`, since...

```python
>>> os.path.split('/test/')
('/test', '')
```

Reply:
Yes, I was also confused by this :smile: After some digging it made sense though.

When I point my browser at `http://localhost:8000/static`, `scope["path"]` is set to `/`.  The path is then transformed by the following line:

`path = os.path.normpath(os.path.join(*scope["path"].split("/")))`

`os.path.join(*scope["path"].split("/"))` produces an empty string.  
Calling `os.path.normpath` on empty string actually returns `'.'`

So in the case of the URL `http://localhost:8000/static`, when `path = os.path.join(self.directory, path)` is called the resulting value for `path` becomes `static/.`